coredb-operator/src/controller.rs

```diff
@@ -139,6 +147,50 @@ impl CoreDB {
             .map_err(Error::KubeError)?;
         Ok(Action::await_change())
     }
+
+    async fn primary_pod(&self, client: Client) -> Result<Pod> {
+        let sts = stateful_set_from_cdb(self);
+        let sts_name = sts.metadata.name.unwrap();
+        let sts_namespace = sts.metadata.namespace.unwrap();
+        let label_selector = format!("statefulset={}", sts_name);
+        let list_params = ListParams::default().labels(&label_selector);
+        let pods: Api<Pod> = Api::namespaced(client, &sts_namespace);
+        let pods = pods.list(&list_params);
+        // For the time being, we assume that the first pod is the primary
+        let primary = pods.await.unwrap().items[0].clone();
+        return Ok(primary);
+    }
+
+    pub async fn psql(&self, command: String, database: String, client: Client) -> Result<String> {
+        let psql_command = vec!["psql", &database, "-c", &command];
+        let attach_params = AttachParams {
+            container: None,
+            tty: false,
+            stdin: true,
+            stdout: true,
+            stderr: true,
+            max_stdin_buf_size: Some(1024),
+            max_stdout_buf_size: Some(1024),
+            max_stderr_buf_size: Some(1024),
+        };
+        let pod_name = self
+            .primary_pod(client.clone())
+            .await
+            .unwrap()
+            .metadata
+            .name
+            .unwrap();
+        let pods: Api<Pod> = Api::namespaced(client.clone(), &self.metadata.namespace.clone().unwrap());
+        let mut attached_process = pods.exec(&pod_name, psql_command, &attach_params).await.unwrap();
+        // attached_process.join();
+        let mut stdout_reader = attached_process.stdout().unwrap();
+        let mut result_stdout = String::new();
+        stdout_reader.read_to_string(&mut result_stdout).await.unwrap();
+        let mut stderr_reader = attached_process.stderr().unwrap();
+        let mut result_stderr = String::new();
+        stderr_reader.read_to_string(&mut result_stderr).await.unwrap();
+        return Ok(result_stdout + &result_stderr);
```

Comment:
I think the return type can be further improved, but I need to learn more rust. would be better if the result is like

```
result.stdout
result.stderr
result.exit_code
```

Reply:
at the moment it's just concatenating stdout and stderr